packages/app-builder-lib/templates/linux/after-install.tpl

```diff
@@ -1,7 +1,11 @@
 #!/bin/bash
 
 # Link to the binary
-ln -sf '/opt/${sanitizedProductName}/${executable}' '/usr/bin/${executable}'
+if type update-alternatives 2>/dev/null >&1; then
+   update-alternatives --install '/usr/bin/${executable}' '${executable}' '/opt/${sanitizedProductName}/${executable}' 100
```

Comment:
I just read this in your changeset. (My apologies for pinging back n forth on this)
> If you are having issues on install or get errors about `/usr/bin/${executable}` already
exists, then simply remove the link and `update-alternatives` will take over from there.

Do we need to update this code as follows? We can't expect all linux users to know they need to remove the link on their own during the install process
```
previousLink = "/usr/bin/${executable}"
# remove previously existing symlink if applicable
[ -L "${previousLink}" ] && [ -e "${previousLink}" ] && rm -f "${previousLink}"

update-alternatives --install "${previousLink}" "${executable}" "/opt/${sanitizedProductName}/${executable}" 100
```

Note: I didn't test this logic, just pulled resources from online and changed `'` to `"`.  So you'd need to test this change

What are your thoughts?

Reply:
No worries, it was an upstream change I wanted to propose into this project and grateful for the thoughtful discussion!

> Do we need to update this code as follows? We can't expect all linux users to know they need to remove the link on their own during the install process

I think I see what you mean in the updates here, so I updated the changeset accordingly. This is very valid and for a project as core as this to so many others, it would cause a lot of heartburn.

In this corrected way, the end-user nor middleware have to take any action in consuming this update.

With the above proposed change, it might produce output durring the install about having to correct a broken link as such:

    update-alternatives: warning: forcing reinstallation of alternative /opt/${sanitizedProductName}/${executable} because link group ${executable} is broken


on each update.

Single quotes are considered string literals. Double-quotes are considered interpolatable strings. Since the templates are passing through a template processor and not the bash interpreter itself, single quotes in this case would be fine since it's not in-script variables that are being used. Since those variables are "userland" variables outside of the script (interpolated by the template filter), single quotes would also be more secure in this case, so I think it might have been correct. In-script variable use might require double-quotes.
(sorry if this sounds wordy -- I am an instructor on my TikTok & YouTube channels :D )